client/domain/PrerequisiteEvalResultObjectFactory.js

```diff
@@ -0,0 +1,52 @@
+// Copyright 2017 The TIE Authors. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS-IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+/**
+ * @fileoverview Factory for creating new frontend instances of
+ * PreRequisiteCheckResult domain objects.
+ */
+
+tie.factory('PrerequisiteEvalResultObjectFactory', [
+  function() {
+    var PrerequisiteEvalResult = function(
+      code, prereqErrorMessage, starterCode) {
+      this._code = code;
+      this._prereqErrorMessage = prereqErrorMessage;
```

Comment:
This is a bit confusing. The purpose of this domain object would presumably be to store enough information to reconstruct the result of the prerequisite check.

So, presumably the only field here should be a list called something like this._prerequisiteCheckFailures. Each entry is a dict with a 'type' key (whose value is PREREQ_CHECK_TYPE_BAD_IMPORT = 'badImport' or PREREQ_CHECK_TYPE_MISSING_STARTER_CODE = 'missingStarterCode' -- these named constants can be declared in app.js). If the key is PREREQ_CHECK_TYPE_MISSING_STARTER_CODE then the dict also has an extra 'expectedStarterCode' key; you may need something similar for badImport as well.

The error messages themselves should presumably go in the service which generates the feedback objects based on the essential info in this domain object.